shared/src/business/useCases/getAllUsersByRoleInteractor.ts

```diff
@@ -0,0 +1,21 @@
+import {
+  ROLE_PERMISSIONS,
+  isAuthorized,
+} from '@shared/authorization/authorizationClientService';
+import { UnauthorizedError } from '@web-api/errors/errors';
+
+export const getAllUsersByRoleInteractor = async (
+  applicationContext: IApplicationContext,
+  { roles }: { roles: string[] },
+) => {
+  const authorizedUser = applicationContext.getCurrentUser();
+  if (!isAuthorized(authorizedUser, ROLE_PERMISSIONS.CREATE_TRIAL_SESSION)) {
```

Comment:
Is it possible to move the authorization check to either its own action (function) or part of the validation action earlier in the sequence its been used?  that way this interactor is cleaner and agnostic to the user fetching roles. 

This interactor seems like an interactor that'll be used in context of other users performing different tasks when fetching roles of users, but please let me know if my assumptions are incorrect.

Reply:
That would cause security risks. Moving validation to the front end where people/hackers can change the behavior as they please will allow people/hackers to gain access to private IRS Practitioners contact information. 

As of now this endpoint is only required to those that can access Create/Update a trial session and as an MVP that is the only Permission this endpoint is required until more features in the future require it we will expand that permission.

Reply:
Thanks for explaining the security risks and I understand it even more. This is all backend so it makes sense to add it here. Thank you.